src/main.rs

```diff
@@ -103,18 +103,27 @@ fn main() -> Result<()> {
 }
 
 fn try_connect(opt: &Opt) -> elephantry::Result<elephantry::Pool> {
-    let mut config = opt.clone().into();
-
-    match elephantry::Pool::from_config(&config) {
-        Ok(client) => Ok(client),
-        Err(err) => {
-            if opt.password && &format!("{}", err) == "invalid configuration: password missing" {
-                let password = rpassword::prompt_password_stdout("Password: ").unwrap();
-                config.password = Some(password.trim_matches('\n').to_string());
-
-                elephantry::Pool::from_config(&config)
-            } else {
-                Err(err)
+    if let Ok(client) = elephantry::Pool::new(&opt.dbname.clone().expect("No database name given"))
+    {
+        Ok(client)
+    } else {
+        {
```

Comment:
There is an extra indent level, you can remove it.

Reply:
Removed.